server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -841,12 +870,12 @@ public BigInteger parse(XContentParser parser, boolean coerce) throws IOExceptio
             }
 
             @Override
-            public Query termQuery(String field, Object value) {
+            public Query termQuery(String field, Object value, Boolean hasDocValues) {
                 if (hasDecimalPart(value)) {
                     return Queries.newMatchNoDocsQuery("Value [" + value + "] has a decimal part");
                 }
                 BigInteger v = parse(value, true);
-                return BigIntegerPoint.newExactQuery(field, v);
+                return UNSIGNED_LONG.rangeQuery(field, v, v, true, true, hasDocValues, null);
```

Comment:
You are in the same class, not need to use UNSIGNED_LONG here:

```suggestion
                return rangeQuery(field, v, v, true, true, hasDocValues, null);
```